.github/workflows/ccpp.yml

```diff
@@ -38,40 +38,25 @@ jobs:
         uses: actions/setup-python@v5
         with:
           python-version: '3.12'
+
       - name: Install linting tools
         run: pip install clang-format cmake-format[YAML] mypy ruff
-      - name: ruff .py files in C++ code
-        run: |
-          cd cpp/
-          ruff check .
-          ruff format --check .
-      - name: ruff Python interface checks
-        run: |
-          cd python/
-          ruff check .
-          ruff format --check .
-      - name: mypy checks
+
+      - name: Run ruff
+        run: ruff check python/ cpp/
```

Comment:
The previous setup was deliberate. It checked formatting, and additionally the split allows the user to quickly see in the CI log which part of the Python interface is failing.